File: UVa/java/UVa_00573_TheSnail.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int height, initialClimbDistance, nightSlide, fatiguePercentage, day;
        double currClimbDistance, currHeight, dailyDrop;
        boolean stop;
        height = scanner.nextInt();

        while(height > 0){
            stop = false;
            day = 1;
            currHeight = 0;
            initialClimbDistance = scanner.nextInt();
            nightSlide = scanner.nextInt();
            fatiguePercentage = scanner.nextInt();
            currClimbDistance = initialClimbDistance;
            dailyDrop = initialClimbDistance*(fatiguePercentage/100.0);

            while(!stop) {
                currHeight += currClimbDistance;

                if(currHeight > height) {
                    stop = true;
                }else{
                    currHeight -= nightSlide;
                    currClimbDistance -= dailyDrop;
                    currClimbDistance = currClimbDistance < 0 ? 0 : currClimbDistance;

                    if(currHeight < 0) {
                        stop = true;
                    }
                }

                if(currHeight <= height && currHeight >= 0){
                    ++day;
                }
            }

            if(currHeight < 0) System.out.println("failure on day " + day);
            else System.out.println("success on day " + day);

            height = scanner.nextInt();
        }
    }
}

File: UVa/kotlin/UVa_00573_TheSnail.kt
import java.util.Scanner

fun main() {
    val scanner = Scanner(System.`in`)
    
    var goal: Int = scanner.nextInt()
    var climbDistance: Int = 0
    var nightDrop: Int = 0
    var dropFactor: Int = 0
    var factoredDrop: Double = 0.0
    var currentHeight: Double = 0.0
    var day = 0
    var currentClimbDistance: Double = 0.0

    while(scanner.hasNext() && goal != 0) {
        climbDistance = scanner.nextInt()
        nightDrop = scanner.nextInt()
        dropFactor = scanner.nextInt()
        
        factoredDrop = climbDistance.toDouble() * (dropFactor.toDouble()/100.0)
        day = 0
        currentHeight = 0.0

        while(currentHeight <= goal && currentHeight >= 0) {
            currentClimbDistance = climbDistance - day*factoredDrop
            if (currentClimbDistance > 0) {
                currentHeight += currentClimbDistance
            }
            day += 1
            if (currentHeight <= goal) {
                currentHeight -= nightDrop
            }
        }

        if (currentHeight < 0) {
            println("failure on day $day")
        } else {
            println("success on day $day")
        }
        goal = scanner.nextInt()
    }
    
}
